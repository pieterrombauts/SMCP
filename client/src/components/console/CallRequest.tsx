import React from 'react';
import styled from 'styled-components';
import Toast from 'react-bootstrap/Toast';

interface CallReqProps {
  className?: string;
  sender: "spartan" | "cronus" | "ethos" | "flight" | "capcom" | "bme";
  show: boolean;
  onHide: (sender: "spartan" | "cronus" | "ethos" | "flight" | "capcom" | "bme") => void;
};

const UCallRequest: React.FC<CallReqProps> = ( props ) => { 
  function handleClose() {
    props.onHide(props.sender);
  }
  return (
    <Toast show={props.show} onClose={handleClose} className={props.className}>
      <Toast.Header closeButton={true}>
        <strong>Call request from {props.sender.toUpperCase()}</strong>
      </Toast.Header>
      <Toast.Body>
        {props.sender.toUpperCase()} is requesting a call from you. Please unmute them in Discord!
      </Toast.Body>
    </Toast>
  );
}

export const CallRequest = styled(UCallRequest)`
  .toast-header {
    justify-content: space-between
  }
`;
