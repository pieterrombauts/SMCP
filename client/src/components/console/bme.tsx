import React from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import Tabs from 'react-bootstrap/Tabs'
import Tab from 'react-bootstrap/Tab'
import bme_1 from "./../../media/Bme_1.png"
import bme_2 from "./../../media/Bme_2.png"

interface AppProps {};

const bme: React.FC<AppProps> = () => {
  return(
         <Tabs defaultActiveKey="Spartan_console" id="uncontrolled-tab-example"  >
        
          <Tab eventKey="BME_1" title="Astronaut Health">
            <img src={bme_1} style={{width:"300px"}}/> 
          </Tab>

          <Tab eventKey="BME_2" title="Astronaut Vitals">
          <img src={bme_2} style={{width:"300px"}}/> 

          </Tab>
        </Tabs>
  );
}

export const Bme = bme;