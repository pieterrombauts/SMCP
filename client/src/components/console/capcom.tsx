import React from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import Tabs from 'react-bootstrap/Tabs'
import Tab from 'react-bootstrap/Tab'
import capcom_1 from "./../../media/Capcom_1.png"

interface AppProps {};

const capcom: React.FC<AppProps> = () => {
  return(
         <Tabs defaultActiveKey="Spartan_console" id="uncontrolled-tab-example" style={{ float: "right"}} >
        
          <Tab eventKey="CAPCOM_1" title="Script">
            <img src={capcom_1} style={{width:"300px"}}/> 
          </Tab>

         
        </Tabs>
  );
}

export const Capcom = capcom;