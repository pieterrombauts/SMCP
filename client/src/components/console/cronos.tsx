import React from 'react';
import styled from 'styled-components';
import 'bootstrap/dist/css/bootstrap.min.css';
import Tabs from 'react-bootstrap/Tabs'
import Tab from 'react-bootstrap/Tab'
import cronos_1 from "./../../media/Cronos_1.png"
import cronos_2 from "./../../media/Cronos_2.png"
import cronos_3 from "./../../media/Cronos_3.png"
import cronos_4 from "./../../media/Cronos_4.png"
import cronos_5 from "./../../media/Cronos_5.png"

interface AppProps {};

const cronos: React.FC<AppProps> = () => {
  return(
         <Tabs  defaultActiveKey="Cronos_console" id="uncontrolled-tab-example" style={{ float: "right"}} >
        
          <Tab eventKey="CRONOS_1" title="Computer Network">
            <img src={cronos_1} style={{width:"300px"}}/> 
          </Tab>

          <Tab eventKey="CRONOS_2" title="UHF Communication">
            <img src={cronos_2} style={{width:"300px"}}/> 
          </Tab>

          <Tab eventKey="CRONOS_3" title="S-Band Communication">
            <img src={cronos_3} style={{width:"300px"}}/> 
          </Tab>

          <Tab eventKey="CRONOS_4" title="Video Communication">
            <img src={cronos_4} style={{width:"300px"}}/> 
          </Tab>

          <Tab eventKey="CRONOS_5" title="Video Communication 2">
            <img src={cronos_5} style={{width:"300px"}}/> 
          </Tab>

         
        </Tabs>
  );
}

export const Cronos = styled(cronos)`
  #uncontrolled-tab-example {
    width: 200px;
  }
`;