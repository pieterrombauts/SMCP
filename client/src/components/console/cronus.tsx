import React from 'react';
import styled from 'styled-components';
import Nav from 'react-bootstrap/Nav'
import Tab from 'react-bootstrap/Tab'
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'
import cronus_1 from "./../../media/Cronus_1.png"
import cronus_2 from "./../../media/Cronus_2.png"
import cronus_3 from "./../../media/Cronus_3.png"
import cronus_4 from "./../../media/Cronus_4.png"
import cronus_5 from "./../../media/Cronus_5.png"

interface AppProps {
  className?: string;
};

const cronus: React.FC<AppProps> = (props) => {
  return (
    <Tab.Container id="cronus-tabs" defaultActiveKey="comp-net">
      <Row className={props.className}>
        <Col id="cronus-content" sm={10}>
          <Tab.Content>
            <Tab.Pane eventKey="comp-net">
              <img src={cronus_1} style={{ width: "1200px" }} />
            </Tab.Pane>
            <Tab.Pane eventKey="uhf-comms">
            <img src={cronus_2} style={{width:"1200px",position:"absolute"}}/>  
              <svg width="1000" height="1000" style={{position:"absolute"}}>
                
                <rect x="50" y="350" rx="5" ry="5" width="460" height="182" style={{fill:"#36485b",stroke:"#2a3a3f",strokeWidth:"5"}} />
                
                <text x="6%" y="38%" fill="white" font-size="25" font-family="Verdana">Space to Space Station</text>

                <text x="8%" y="43%" fill="white" font-size="20" font-family="Verdana">Radio1 Power</text>
                <text x="25%" y="43%" fill="#d6ab62" font-size="20" font-family="Verdana">Off</text>

                <text x="8%" y="46%" fill="white" font-size="20" font-family="Verdana">Radio1 Power</text>
                <text x="25%" y="46%" fill="#d6ab62" font-size="20" font-family="Verdana">Off</text>

                <text x="6%" y="51%" fill="white" font-size="25" font-family="Verdana">UHF Sync Lcok</text>
                <text x="28%" y="51%" fill="#d6ab62" font-size="20" font-family="Verdana">Frame Sync</text>
                
              </svg>

              <svg width="100%" height="100%" style={{position:"absolute"}}>
                
                <rect x="740" y="370" rx="5" ry="5" width="390" height="135" style={{fill:"#36485b",stroke:"#2a3a3f",strokeWidth:"5"}} />
                
                <text x="62%" y="46%" fill="white" font-size="20" font-family="Verdana">IAC1 Power</text>
                <text x="75%" y="46%" fill="#d6ab62" font-size="20" font-family="Verdana">Active</text>

                <text x="62%" y="49%" fill="white" font-size="20" font-family="Verdana">IAC2 Power</text>
                <text x="75%" y="49%" fill="#d6ab62" font-size="20" font-family="Verdana">Active</text>

              </svg>
            </Tab.Pane>
            
            <Tab.Pane eventKey="s-band-comms">
            <img src={cronus_3} style={{width:"1200px",position:"absolute"}}/>  
              <svg width="1000" height="1000" style={{position:"absolute"}}>
                
                <rect x="415" y="265" rx="5" ry="5" width="415" height="385" style={{fill:"#36485b",stroke:"#2a3a3f",strokeWidth:"5"}} />
                
                <text x="42.5%" y="31.5%" fill="white" font-size="45" font-family="Verdana">S-Band Status</text>

                <text x="42.5%" y="35%" fill="white" font-size="20" font-family="Verdana">Active S-Band</text>
                <text x="65%" y="35%" fill="#d6ab62" font-size="20" font-family="Verdana">1</text>

                <text x="42.5%" y="38%" fill="white" font-size="20" font-family="Verdana">S-Band1 Power</text>
                <text x="65%" y="38%" fill="#d6ab62" font-size="20" font-family="Verdana">On</text>

                <text x="42.5%" y="41%" fill="white" font-size="20" font-family="Verdana">Elevation Gimbal</text>
                <text x="65%" y="41%" fill="#d6ab62" font-size="20" font-family="Verdana">59.14</text>
                <text x="72%" y="41%" fill="white" font-size="20" font-family="Verdana">°</text>

                <text x="42.5%" y="44%" fill="white" font-size="20" font-family="Verdana">Azimuth Gimbal</text>
                <text x="65%" y="44%" fill="#d6ab62" font-size="20" font-family="Verdana">1.59</text>
                <text x="72%" y="44%" fill="white" font-size="20" font-family="Verdana">°</text>


                <text x="42.5%" y="47%" fill="white" font-size="20" font-family="Verdana">S-Band2 Power</text>
                <text x="65%" y="47%" fill="#d6ab62" font-size="20" font-family="Verdana">Off</text>

                <text x="42.5%" y="50%" fill="white" font-size="20" font-family="Verdana">Elevation Gimbal</text>
                <text x="65%" y="50%" fill="#d6ab62" font-size="20" font-family="Verdana">0.00</text>
                <text x="72%" y="50%" fill="white" font-size="20" font-family="Verdana">°</text>


                <text x="42.5%" y="53%" fill="white" font-size="20" font-family="Verdana">Azimuth Gimbal</text>
                <text x="65%" y="53%" fill="#d6ab62" font-size="20" font-family="Verdana">0.00</text>
                <text x="72%" y="53%" fill="white" font-size="20" font-family="Verdana">°</text>
              </svg>
              
            </Tab.Pane>

            <Tab.Pane eventKey="vid-comms-1">
              <img src={cronus_4} style={{ position:"absolute", width: "1200px" }} />
              <svg width="1000" height="1000" style={{position:"absolute"}}>
                
                <rect x="415" y="265" rx="5" ry="5" width="405" height="375" style={{fill:"#36485b",stroke:"#2a3a3f",strokeWidth:"5"}}/>
                
                <text x="42.5%" y="31.5%" fill="white" font-size="45" font-family="Verdana">Ku-Band Status</text>

                <text x="42.5%" y="35%" fill="white" font-size="20" font-family="Verdana">Downlink 1 Status</text>
                <text x="65%" y="35%" fill="#d6ab62" font-size="20" font-family="Verdana">ACTIVE</text>

                <text x="42.5%" y="38%" fill="white" font-size="20" font-family="Verdana">Downlink 2 Status</text>
                <text x="65%" y="38%" fill="#d6ab62" font-size="20" font-family="Verdana">ACTIVE</text>

                <text x="42.5%" y="41%" fill="white" font-size="20" font-family="Verdana">Downlink 3 Status</text>
                <text x="65%" y="41%" fill="#d6ab62" font-size="20" font-family="Verdana">ACTIVE</text>

                <text x="42.5%" y="44%" fill="white" font-size="20" font-family="Verdana">Downlink 4 Status</text>
                <text x="65%" y="44%" fill="#d6ab62" font-size="20" font-family="Verdana">ACTIVE</text>


                <text x="42.5%" y="55%" fill="white" font-size="20" font-family="Verdana">Elevation Gimbal</text>
                <text x="67%" y="55%" fill="#d6ab62" font-size="20" font-family="Verdana">-48.85</text>
                <text x="75%" y="55%" fill="white" font-size="20" font-family="Verdana">°</text>

                <text x="42.5%" y="58%" fill="white" font-size="20" font-family="Verdana">Cross-Elevation Gimbal</text>
                <text x="67%" y="58%" fill="#d6ab62" font-size="20" font-family="Verdana">36.39</text>
                <text x="75%" y="58%" fill="white" font-size="20" font-family="Verdana">°</text>
              </svg>

            </Tab.Pane>
            
            
            <Tab.Pane eventKey="vid-comms-2">
            <img src={cronus_5} style={{ position:"absolute", width: "1200px" }} />
              <svg width="1000" height="1000" style={{position:"absolute"}}>
                
                <rect x="570" y="550" rx="5" ry="5" width="370" height="160" style={{fill:"#36485b",stroke:"#2a3a3f",strokeWidth:"5"}}/>
                
                <text x="58%" y="58%" fill="white" font-size="17" font-family="Verdana">1)</text>
                <text x="61%" y="58%" fill="#d6ab62" font-size="20" font-family="Verdana">P1 Upper Outboard (P1UPOB)</text>

                <text x="58%" y="62%" fill="white" font-size="17" font-family="Verdana">1)</text>
                <text x="61%" y="62%" fill="#d6ab62" font-size="20" font-family="Verdana">P1 Lower Outboard (P1LOOB)</text>

                <text x="58%" y="66%" fill="white" font-size="17" font-family="Verdana">1)</text>
                <text x="61%" y="66%" fill="#d6ab62" font-size="20" font-family="Verdana">Labs Starboard (LABS)</text>

                <text x="58%" y="70%" fill="white" font-size="17" font-family="Verdana">1)</text>
                <text x="61%" y="70%" fill="#d6ab62" font-size="20" font-family="Verdana">0E</text>

                
                
              </svg>
            </Tab.Pane>
          
          </Tab.Content>
        </Col>
        <Col id="cronus-buttons" sm={2}>
          <Nav variant="pills" className="flex-column">
            <Nav.Item>
              <Nav.Link eventKey="comp-net">COMPUTER NETWORK</Nav.Link>
            </Nav.Item>
            <Nav.Item>
              <Nav.Link eventKey="uhf-comms">UHF COMMS</Nav.Link>
            </Nav.Item>
            <Nav.Item>
              <Nav.Link eventKey="s-band-comms">S-BAND COMMS</Nav.Link>
            </Nav.Item>
            <Nav.Item>
              <Nav.Link eventKey="vid-comms-1">VIDEO COMMS 1</Nav.Link>
            </Nav.Item>
            <Nav.Item>
              <Nav.Link eventKey="vid-comms-2">VIDEO COMMS 2</Nav.Link>
            </Nav.Item>
          </Nav>
        </Col>
      </Row>
    </Tab.Container>

    //  <Tabs  defaultActiveKey="Cronus_console" id="uncontrolled-tab-example" style={{ float: "right"}} >

    //   <Tab eventKey="CRONUS_1" title="Computer Network">
    //     <img src={cronus_1} style={{width:"300px"}}/> 
    //   </Tab>

    //   <Tab eventKey="CRONUS_2" title="UHF Communication">
    //     <img src={cronus_2} style={{width:"300px"}}/> 
    //   </Tab>

    //   <Tab eventKey="CRONUS_3" title="S-Band Communication">
    //     <img src={cronus_3} style={{width:"300px"}}/> 
    //   </Tab>

    //   <Tab eventKey="CRONUS_4" title="Video Communication">
    //     <img src={cronus_4} style={{width:"300px"}}/> 
    //   </Tab>

    //   <Tab eventKey="CRONUS_5" title="Video Communication 2">
    //     <img src={cronus_5} style={{width:"300px"}}/> 
    //   </Tab>


    // </Tabs>
  );
}

export const Cronus = styled(cronus)`
  width: 100%;
  height: 100%;
  position: absolute;

  #cronus-buttons {
    display: flex;
    flex-direction: column;
    justify-content: center;
  }

  #cronus-buttons .nav {
    margin-left: 20px;
    max-width: 300px;
    background-color: #f8f9fa;
    border-radius: 3px;
  }

  #cronus-buttons .nav-item a {
    padding: 20px 30px;
  }

`;