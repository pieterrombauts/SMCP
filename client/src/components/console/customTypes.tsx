export interface ethosLifeSupportDisplay {
  tranquility: {
    pressure: number,
    temp: number,
    ppO2: number,
    ppN2: number,
    ppCO2: number
  },
  destiny: {
    pressure: number,
    temp: number,
    ppO2: number,
    ppN2: number,
    ppCO2: number
  },
  harmony: {
    pressure: number,
    coolant_temp: number
  },
  quest: {
    pressure: number,
    crewlock_pressure: number,
    EVA_O2: number,
    N2: number,
    O2: number
  }
}

export interface ethosThermalSystemDisplay
{
  destinyLab: {
    lowTLoop: number;
    moderateTLoop: number,
    lowTemp: number;
    moderateTemp: number
  },

  harmonyNode : {
    lowTLoop: number;
    moderateTLoop: number,
    lowTemp: number;
    moderateTemp: number
  }

  tranquilityNode : {
    lowTLoop: number;
    moderateTLoop: number,
    lowTemp: number;
    moderateTemp: number
  }

}

export interface ethosRegenLifeSupportDisplay {
  upa: {
    urineTankLevel: number
  }

  wpa: { 
    wastewaterTankLevel: number,
    cleanWaterTankLevel: number
  }

  oga: {
    oxygenProductionRate: number
  }
}

export interface cronusCompNetDisplay {
  compNet: {
    time: string,
    standardCmds: number,
    loadCmds: number
  },
}

export interface cronusSBandDisplay {
  sBand: {
    elevationGimbal: number,
    azimuthGimbal: number
  }

}

export interface cronusVComm1Display {
  vComms1: {
    elevationGimbal: number,
    crossElevationGimbal: number
  }
}

export interface statusReport {
  _ID: string,
  _sender: string,
  _subject: string,
  _content: string,
  _status: string,
  _time: string,
  _comments: string[]
}

export interface bmeVitalSignsDisplay {
  astronaut1: {
    heartRate: number,
    heartRateMax: number,
    bloodPressureSystolic: number,
    bloodPressureDiastolic: number,
  }

  astronaut2: {
    heartRate: number,
    heartRateMax: number,
    bloodPressureSystolic: number,
    bloodPressureDiastolic: number,
  }

  astronaut3: {
    heartRate: number,
    heartRateMax: number,
  }

  astronaut4: {
    heartRate: number,
    heartRateMax: number,
  }
}

export interface bmeVitalSignsTemps {
  astronaut1: {
    temperature: number
  }

  astronaut2: {
    temperature: number
  }

  astronaut3: {
    temperature: number
  }

  astronaut4: {
    temperature: number
  }
}

export interface bmeEVASuitDisplay {
  eva1: {
    helmetPressure: number,
    o2: number,
    water: number,
    batteryCharge: number
  },

  eva2: {
    helmetPressure: number,
    o2: number,
    water: number,
    batteryCharge: number
  }
}

export interface consoleUsers {
  spartan: string[],
  cronus: string[],
  ethos: string[],
  flight: string[],
  capcom: string[],
  bme: string[]
}

export interface firstConsoleOpens {
  spartan: boolean,
  cronus: boolean,
  ethos: boolean,
  flight: boolean,
  capcom: boolean,
  bme: boolean,
  efn: boolean,
  ostpv: boolean
}

export type OptionType = {
  value: string;
  label: string;
}

export interface event {
  id: number,
  resourceId: string,
  astronaut: string,
  title: string,
  start: string,
  end: string,
  procedure: string,
  location: string,
  classNames: string[]
}

export interface selectedEvent {
  id: number,
  title: string,
  start: string,
  end: string,
  extendedProps: {
    astronaut: string,
    procedure: string,
    location: string,
  }
  classNames: string[]
}

export interface tableEntry {
  eventNum: string,
  annun: string,
  cl: string,
  ack: boolean,
  sys: string,
  message: string
}

export interface chartData {
  datasets: {
      data: {x: string, y: number}[],
      backgroundColor: string,
      borderColor: string,
      borderWidth: number,
      fill: boolean,
    }[]
}