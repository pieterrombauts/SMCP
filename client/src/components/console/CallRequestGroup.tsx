import React from 'react'
import styled from 'styled-components'
import { CallRequest } from './CallRequest'

interface CallReqGroupProps {
  className?: string;
  callRequests: {
    spartan: boolean;
    cronus: boolean;
    ethos: boolean;
    flight: boolean;
    capcom: boolean;
    bme: boolean;
  };
  onHide: (sender: "spartan" | "cronus" | "ethos" | "flight" | "capcom" | "bme") => void;
};

const UCallRequestGroup: React.FC<CallReqGroupProps> = ( props ) => { 
  return (
    <div className={props.className}>
        <CallRequest sender="spartan" show={props.callRequests["spartan"]} onHide={props.onHide}/>
        <CallRequest sender="cronus" show={props.callRequests["cronus"]} onHide={props.onHide}/>
        <CallRequest sender="ethos" show={props.callRequests["ethos"]} onHide={props.onHide}/>
        <CallRequest sender="flight" show={props.callRequests["flight"]} onHide={props.onHide}/>
        <CallRequest sender="capcom" show={props.callRequests["capcom"]} onHide={props.onHide}/>
        <CallRequest sender="bme" show={props.callRequests["bme"]} onHide={props.onHide}/>
    </div>
  );
}

export const CallRequestGroup = styled(UCallRequestGroup)`
  position: absolute;
  top: 20px;
  right: 20px;

  .fade:not(.show) {
    display: none;
  }
`;
