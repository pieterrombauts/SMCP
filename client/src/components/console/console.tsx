import React from 'react';
import styled from 'styled-components';
import 'bootstrap/dist/css/bootstrap.min.css';
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'
import Tab from 'react-bootstrap/Tab'
import Nav from 'react-bootstrap/Nav'
import {Spartan} from './spartan';
import {Capcom} from './capcom';
import {Cronus1} from './cronus/cronus';
import {Ethos} from './ethos/ethos';
import {Flight} from './flight';
import {Bme} from './bme';

interface AppProps {
  className?: string;
};

const UConsole: React.FC<AppProps> = ( props ) => {
  return(
    <Tab.Container id="console-tabs">
      <Row className={props.className}>
        <Col id="console-buttons" sm={2}>
          <Nav variant="pills" className="flex-column">
            <Nav.Item>
              <Nav.Link eventKey="spartan">SPARTAN</Nav.Link>
            </Nav.Item>
            <Nav.Item>
              <Nav.Link eventKey="cronus">CRONUS</Nav.Link>
            </Nav.Item>
            <Nav.Item>
              <Nav.Link eventKey="ethos">ETHOS</Nav.Link>
            </Nav.Item>
            <Nav.Item>
              <Nav.Link eventKey="flight">FLIGHT</Nav.Link>
            </Nav.Item>
            <Nav.Item>
              <Nav.Link eventKey="capcom">CAPCOM</Nav.Link>
            </Nav.Item>
            <Nav.Item>
              <Nav.Link eventKey="bme">BME</Nav.Link>
            </Nav.Item>
          </Nav>
        </Col>
        <Col id="console-content" sm={10}>
          <Tab.Content>
            <Tab.Pane eventKey="spartan">
              <Spartan />
            </Tab.Pane>
            <Tab.Pane eventKey="cronus">
              <Cronus1 />
            </Tab.Pane>
            <Tab.Pane eventKey="ethos">
              <Ethos />
            </Tab.Pane>
            <Tab.Pane eventKey="flight">
              <Flight />
            </Tab.Pane>
            <Tab.Pane eventKey="capcom">
              <Capcom />
            </Tab.Pane>
            <Tab.Pane eventKey="bme">
              <Bme />
            </Tab.Pane>
          </Tab.Content>
        </Col>
      </Row>
    </Tab.Container>
  );
}

export const Console = styled(UConsole)`
  width: 100%;
  height: 100%;
  position: absolute;
  z-index: 1;
  #console-buttons {
    display: flex;
    flex-direction: column;
    justify-content: center;
  }
  #console-buttons .nav {
    margin-left: 20px;
    max-width: 150px;
    background-color: #f8f9fa;
    border-radius: 3px;
  }
  #console-buttons .nav-item a {
    padding: 20px 30px;
  }
`;