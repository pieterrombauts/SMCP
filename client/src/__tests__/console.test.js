import React from "react";
import 'jest-styled-components'
import toJson from 'enzyme-to-json'
import { UConsole } from '../components/console/console'
import { shallow, mount, render} from 'enzyme';

describe("Overall console display", () => {
  it('renders correctly', () => {
    const wrapper = shallow(<UConsole />)
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  })
})