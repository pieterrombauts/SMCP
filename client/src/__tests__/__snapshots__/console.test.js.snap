// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overall console display renders correctly 1`] = `
<div
  id="game-container"
>
  <TabContainer
    id="console-tabs"
  >
    <Row
      noGutters={false}
    >
      <Col
        id="console-buttons"
        sm={2}
      >
        <Nav
          className="flex-column"
          fill={false}
          justify={false}
          variant="pills"
        >
          <NavItem>
            <NavLink
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
              eventKey="spartan"
              onClick={[Function]}
            >
              SPARTAN
            </NavLink>
          </NavItem>
          <NavItem>
            <NavLink
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
              eventKey="cronus"
              onClick={[Function]}
            >
              CRONUS
            </NavLink>
          </NavItem>
          <NavItem>
            <NavLink
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
              eventKey="ethos"
              onClick={[Function]}
            >
              ETHOS
            </NavLink>
          </NavItem>
          <NavItem>
            <NavLink
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
              eventKey="flight"
              onClick={[Function]}
            >
              FLIGHT
            </NavLink>
          </NavItem>
          <NavItem>
            <NavLink
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
              eventKey="capcom"
              onClick={[Function]}
            >
              CAPCOM
            </NavLink>
          </NavItem>
          <NavItem>
            <NavLink
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
              eventKey="bme"
              onClick={[Function]}
            >
              BME
            </NavLink>
          </NavItem>
        </Nav>
        <Button
          active={false}
          className=""
          disabled={false}
          onClick={[Function]}
          type="button"
          variant="outline-primary"
        >
          OSTPV
        </Button>
        <Button
          active={false}
          className=""
          disabled={false}
          onClick={[Function]}
          type="button"
          variant="outline-primary"
        >
          EFN
        </Button>
      </Col>
      <Col
        id="console-content"
        sm={10}
      >
        <ForwardRef
          id="tab-content"
        >
          <TabPane
            eventKey="spartan"
          >
            <Styled(USpartan) />
          </TabPane>
          <TabPane
            eventKey="cronus"
          >
            <Connect(Styled(Cronus)) />
          </TabPane>
          <TabPane
            eventKey="ethos"
          >
            <Styled(UEthos) />
          </TabPane>
          <TabPane
            eventKey="flight"
            id="flight-tab"
          >
            <Styled(UFlight)
              handleView={[Function]}
              reports={Array []}
            />
          </TabPane>
          <TabPane
            eventKey="capcom"
          >
            <Connect(Styled(UCapcom)) />
          </TabPane>
          <TabPane
            eventKey="bme"
          >
            <Styled(UBme) />
          </TabPane>
        </ForwardRef>
      </Col>
    </Row>
  </TabContainer>
  <CSSTransition
    classNames="modal"
    in={false}
    onEnter={[Function]}
    onExit={[Function]}
    timeout={500}
  >
    <Styled(UOSTPVModal)
      closeFunction={[Function]}
      show={false}
    />
  </CSSTransition>
  <CSSTransition
    classNames="modal"
    in={false}
    onEnter={[Function]}
    onExit={[Function]}
    timeout={500}
  >
    <Styled(USTATUSModal)
      closeFunction={[Function]}
      show={false}
    />
  </CSSTransition>
  <CSSTransition
    classNames="modal"
    in={false}
    onEnter={[Function]}
    onExit={[Function]}
    timeout={500}
  >
    <Styled(UViewEFNModal)
      closeFunction={[Function]}
      reports={Array []}
      selectedEfn=""
      show={false}
    />
  </CSSTransition>
</div>
`;
